packages/toolpad-components/src/FilePicker.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import {
+  TextField as MuiTextField,
+  TextFieldProps as MuiTextFieldProps,
+  BoxProps,
+} from '@mui/material';
+import { createComponent } from '@mui/toolpad-core';
+
+export type Props = MuiTextFieldProps & {
+  // alignItems?: BoxProps['alignItems'];
+  // justifyContent?: BoxProps['justifyContent'];
+  multiple: boolean;
+};
+
+function FilePicker({ multiple, ...props }: Props) {
+  return <MuiTextField {...props} type="file" value={undefined} inputProps={{ multiple }} />;
+}
+
+interface FullFile {
+  name: string;
+  type: string;
+  size: number;
+  dataURL: null | string;
+  arrayBuffer: null | string | ArrayBuffer;
+  loaded: boolean;
+}
+
+export default createComponent(FilePicker, {
+  layoutDirection: 'both',
+  argTypes: {
+    value: {
+      typeDef: { type: 'file' },
+      onChangeProp: 'onChange',
+      onChangeHandler: (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
@Janpot before I sink more time into this I have a few questions:

1. Do you have any alternative suggestions how we should be passing file value to the user? Right now I'm converting `FileList` object to array which should contain actual file value so we can avoid forcing users to use `FileReader` when binding values. At the moment `FileReader` is not even accessible, but also it would require handling promises in binding editor, which I'm not sure if is supported now?
2. This change handler feels a bit ugly as it's handling `onload` and asynchronously setting values on objects. Seems that up until now we didn't need to handle async value setting via change handlers, wondering if we should consider adding `Promise` handling or keep it for sync values only?
3. Without an actual usecase for file picker it's hard to tell what we need, but wondering if dataURL and array buffer would be enough for now? Or maybe I should use something entirely different here?

Reply:
Similar to what I suggested with the DatePicker, remove `onChangeHandler`, program it in the component itself.

Something along the lines of
```tsx
function FilePicker({ multiple, onFilesChange, ...props }: Props) {
  const handleChange = React.useCallback(async (event) => {
    const files = await Promise.all(Array.from(event.target.files, async (file) => {
      return {
        name: file.name,
        type: file.type,
        base64: await readAsBase64(file)
      }
    }))
    onFilesChange(files)
  }, [onFilesChange]);
  
  return <MuiTextField {...props} type="file" onChange={handleChange} inputProps={{ multiple }} />;
}
```

Reply:
Thanks a lot! I didn't got it the first time (with DatePicker), but now it makes sense 👍